R/zarr-array.R

```diff
@@ -323,38 +322,13 @@ ZarrArray <- R6::R6Class("ZarrArray",
         return(out)
       }
 
-      parallel_option <- getOption("pizzarr.parallel_read_enabled")
-      cl <- parse_parallel_option(parallel_option)
-      is_parallel <- is_truthy_parallel_option(cl)
-      apply_func <- lapply
-      if(is_parallel) {
-        if(!requireNamespace("pbapply", quietly = TRUE)) {
-          stop("Parallel reading requires the 'pbapply' package.")
-        }
-
-        if(is.integer(cl) & .Platform$OS.type == "windows") {
-          # See #105
-          cl <- parallel::makeCluster(cl)
-          on.exit(parallel::stopCluster(cl))
-        }
-        
-        apply_func <- function(X, FUN, ..., cl = NULL) {
-          
-          if(isTRUE(cl == "future")) {
-            pbapply::pblapply(X, FUN, ..., 
-                              future.packages = "Rarr",
-                              future.seed=TRUE, cl = cl)
-          } else {
-            pbapply::pblapply(X, FUN, ..., cl = cl)
-          }
-        }
-
-      }
+      ps <- get_parallel_settings(parallel_option = getOption("pizzarr.parallel_read_enabled", FALSE))
+      if(ps$close) on.exit(try(parallel::stopCluster(ps$cl), silent = TRUE))
 
       parts <- indexer$iter()
-      part1_results <- apply_func(parts, function(proj, cl = NA) {
+      part1_results <- ps$FUN(parts, function(proj, cl = NA) {
```

Comment:
```suggestion
      part1_results <- ps$FUN(parts, function(proj) {
```

It seems like since `cl = cl` has been changed to `cl = ps$cl` below, this `cl = NA` parameter is never used, correct? Do we still need to keep the `cl = NA` in the function signature (here, in the writing case, and within `get_parallel_settings`)?

Reply:
I'll double check this and implement in a follow up commit.